src/controllers/authsController.ts

```diff
@@ -0,0 +1,200 @@
+import Company from "../models/CompanyModel";
+import Officer from "../models/OfficerModel";
+import bcrypt from "bcryptjs";
+import jwt, {Secret} from "jsonwebtoken";
+import {Request, Response, NextFunction} from "express";
+import { Types } from "mongoose";
+
+interface RegisterCompanyType {
+  _id?: Types.ObjectId;
+  name: string;
+  email: string;
+  phoneNumber: string;
+  city: string;
+  state: string;
+  password: string;
+  offices?: Types.ObjectId[],
+  logo?: string;
+  rating?:number;
+  isAdmin?: boolean;
+};
+
+interface CompanyType {
+  _doc: RegisterCompanyType;
+}
+
+interface RegisterOfficerType {
+  _id?: Types.ObjectId;
+  name: string;
+  password: string;
+  address: string;
+  companyId: Types.ObjectId;
+  location: string;
+  phoneNumber: string;
+  pending?: number;
+  viewed?: number;
+  picked?: number;
+  transit?: number;
+  isAdmin?: boolean;
+}
+
+interface OfficerType {
+  _doc: RegisterOfficerType;
+}
+
+const secretKey: Secret = String(process.env.JWT);
+
+export const registerCompany = async(req:Request, res:Response, next:NextFunction) => {
+  const {name, email, phoneNumber, city, state, password, logo}: RegisterCompanyType = req.body;
+  try {
+    const salt = bcrypt.genSaltSync(10);
+    const hash = bcrypt.hashSync(password, salt)
+
+    const newCompany = new Company<RegisterCompanyType>({
+      name,
+      email,
+      phoneNumber,
+      city,
+      state,
+      password: hash,
+    })
+
+    const companyExist: RegisterCompanyType | null = await Company.findOne({name});
```

Comment:
```suggestion
    const companyExist: RegisterCompanyType | null = await Company.findOne({name}).lean();
```
This is what I meant above - [ref](https://mongoosejs.com/docs/tutorials/lean.html).